app/src/test/java/org/simple/clinic/summary/linkId/LinkIdWithPatientUpdateTest.kt

```diff
@@ -0,0 +1,47 @@
+package org.simple.clinic.summary.linkId
+
+import com.spotify.mobius.test.NextMatchers.hasEffects
+import com.spotify.mobius.test.NextMatchers.hasModel
+import com.spotify.mobius.test.NextMatchers.hasNoEffects
+import com.spotify.mobius.test.UpdateSpec
+import com.spotify.mobius.test.UpdateSpec.assertThatNext
+import org.junit.Test
+import org.simple.clinic.TestData
+import org.simple.clinic.patient.businessid.Identifier
+import java.util.UUID
+
+class LinkIdWithPatientUpdateTest {
+  private val updateSpec = UpdateSpec<LinkIdWithPatientModel, LinkIdWithPatientEvent, LinkIdWithPatientEffect>(LinkIdWithPatientUpdate())
+  private val defaultModel = LinkIdWithPatientModel.create()
+  private val patientUuid = UUID.fromString("e2770294-fe11-4f66-ab8d-0da30382d957")
+  private val identifier = TestData.identifier(type = Identifier.IdentifierType.BpPassport, value = "435645")
+
+  @Test
+  fun `when link id with patient view is shown, then get patient id, bp passport number, and get patient name from patient id`() {
+    updateSpec
+        .given(defaultModel)
+        .whenEvent(LinkIdWithPatientViewShown(patientUuid, identifier))
+        .then(
+            assertThatNext(
+                hasModel(defaultModel.linkIdWithPatientViewShown(patientUuid, identifier)),
+                hasEffects(GetPatientNameFromId(patientUuid))
+            )
+        )
+  }
+
+  @Test
+  fun `when patient name is received, then update the patient name`() {
```

Comment:
We can reword this to something like "when the patient details are loaded, then update the UI".